src/SDK/Trace/Behavior/LoggerAwareTrait.php

```diff
@@ -23,20 +27,65 @@ public function setDefaultLogLevel(string $logLevel): void
         $this->defaultLogLevel = $logLevel;
     }
 
+    /**
+     * @return static
+     */
+    public function withLogger(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+
+        return $this;
+    }
+
     /**
      * @param string $message
      * @param array $context
      * @param string|null $level
      */
     protected function log(string $message, array $context = [], ?string $level = null): void
     {
+        //TODO basic info on which class created the message, cheaper than calculating from a stack trace
+        $context['source'] = __CLASS__;
         $this->getLogger()->log(
             $level ?? $this->defaultLogLevel,
             $message,
             $context
         );
     }
 
+    protected function debug(string $message, array $context = []): void
```

Comment:
Could you rename these methods with a `log` prefix (eg. `logDebug`), as these names have no meaning outside of logging.